tools/pre-commit.sh

```diff
@@ -1,28 +1,36 @@
 #!/usr/bin/env bash
+# Copy or link this script into .git/hooks/
+#   $ cp tools/pre-commit.sh .git/hooks/pre-commit
+# It runs automatically in the project root directory (parent of .git/).
 
-python -m black --check ./ > /dev/null 2>&1
-need_format=$?
-set -e
-if [ $need_format -ne 0 ]
-then
-    python -m black ./
-    echo Some Python files were formatted
-    echo You need to do git add and git commit again
-    exit $need_format
+EXIT_STATUS=0
+PROJECT_ROOT=`git rev-parse --show-toplevel`
+
+# Environmental variable used to notify hooks we are running in a workflow.
+ARMORY_CI_TEST="${ARMORY_CI_TEST:-0}"
+
+# Collect tracked files based on ARMORY_CI_TEST
+TRACKED_FILES="git diff HEAD --name-only"
+if [ "${ARMORY_CI_TEST}" -ne 0 ]; then
+    TRACKED_FILES="git ls-files"
 fi
-set +e
-
-python -m tools.format_json --check > /dev/null 2>&1
-need_format=$?
-set -e
-if [ $need_format -ne 0 ]
-then
-    python -m tools.format_json
-    echo Some JSON files were formatted
-    echo You need to do git add and git commit again
-    exit $need_format
+
+
+pushd $PROJECT_ROOT > /dev/null
+    # Source hooks
+    for FILE in `ls -1 ./tools/hooks/*.sh | sort`; do
+        echo "Importing ${FILE}..."
+        source "${FILE}"
+        if [ $? -ne 0 ]; then
+            EXIT_STATUS=1
+        fi
+    done
+popd > /dev/null
+
+
+if [ "${EXIT_STATUS}" -ne 0 ]; then
+  echo "🚨 Pre-commit hooks failed. Please re-run 'git add' to fix the issues 🚑"
```

Comment:
Certain failed hooks cannot be fixed by `git add`. If there are syntax or parsing errors in python or json, for instance, as well as any errors from flake.